common/log.hpp

```diff
@@ -0,0 +1,91 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef MENDER_LOG_HPP
+#define MENDER_LOG_HPP
+
+#include "common_config.h"
+
+#include "loglevel.hpp"
+#include "logfield.hpp"
+
+#ifdef MENDER_LOG_BOOST
+#include <boost/log/common.hpp>
+#include <boost/log/sources/logger.hpp>
+#endif
+
+namespace mender::log {
+
+using namespace std;
+
+void setup();
+
+class Logger {
+private:
+#ifdef MENDER_LOG_BOOST
+	boost::log::sources::severity_logger<LogLevel> logger;
+#endif
+
+	LogLevel _level {LogLevel::Debug};
+
+	string name {};
+
+	void add_field(const LogField &field);
+
+public:
+	Logger(const string name);
+
+	void setLevel(LogLevel level);
+
+	LogLevel level();
+
+	template <typename... Fields>
+	Logger withFields(const Fields &&...fields) {
+		auto l = Logger("Foo");
+		for (const auto f : {fields...}) {
+			l.add_field(f);
+		}
+		return l;
+	}
+
+	void log(LogLevel level, const string message);
```

Comment:
Lots of copying I know, but not sure how this _should_ be done properly.

Overload for all types? References and moves?